### PR TITLE
Fix doc

### LINKED ISSUE #59789: DOC: Intro to data structures
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/user_guide/dsintro.html#basics-series

### Documentation problem

From scalar value
If data is a scalar value, an index must be provided. 

### Suggested fix for documentation

If data is a scalar value, an index is not mandatory. 

image

### PR DESCRIPTION
**Pull request by fbourgey**
- [ ] closes #59789

### CODE REVIEW

File: doc/source/user_guide/dsintro.rst [hunk 1]
```diff
@@ -87,8 +87,8 @@ index will be pulled out.
 
 **From scalar value**
 
-If ``data`` is a scalar value, an index must be
-provided. The value will be repeated to match the length of **index**.
+If ``data`` is a scalar value, the value will be repeated to match
+the length of **index**.
```
Comment (rhshadrach) [inline doc/source/user_guide/dsintro.rst:91]: It seems to me we should also indicate what happens if no `index` is provided.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: doc/source/user_guide/dsintro.rst (hunk 1 above)
Comment (fbourgey) [reply doc/source/user_guide/dsintro.rst:91]: Something like this?

If ``data`` is a scalar value, the value will be repeated to match
the length of **index**. If the **index** is not provided, it defaults to ``RangeIndex (0, 1, 2, …, n)``.

File: doc/source/user_guide/dsintro.rst [hunk 2]
```diff
@@ -88,7 +88,8 @@ index will be pulled out.
 **From scalar value**
 
 If ``data`` is a scalar value, the value will be repeated to match
-the length of **index**.
+the length of **index**.  If the index is not provided, it defaults
+to ``RangeIndex (0, 1, 2, …, n)``.
```
Comment (rhshadrach) [inline doc/source/user_guide/dsintro.rst:92]: `n` here is not defined, and I think in this notation would be `0`, but this is not the notation used to construct a RangeIndex. In particular, `RangeIndex(0)` here is wrong. Can you make this `RangeIndex(1)`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/dsintro.rst (hunk 2 above)
Comment (fbourgey) [reply doc/source/user_guide/dsintro.rst]: I made the change.

File: doc/source/user_guide/dsintro.rst [hunk 3]
```diff
@@ -87,8 +87,9 @@ index will be pulled out.
 
 **From scalar value**
 
-If ``data`` is a scalar value, an index must be
-provided. The value will be repeated to match the length of **index**.
+If ``data`` is a scalar value, the value will be repeated to match
+the length of **index**.  If the **index** is not provided, it defaults
+to ``RangeIndex (1)``.
```
Comment (rhshadrach) [inline doc/source/user_guide/dsintro.rst:92]: Can you remove the space so it is `RangeIndex(1)`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/dsintro.rst (hunk 3 above)
Comment (fbourgey) [reply doc/source/user_guide/dsintro.rst]: done

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**